polyfill/lib/absolute.mjs

```diff
@@ -63,7 +63,7 @@ export class Absolute {
     ns += GetSlot(duration, NANOSECONDS);
     if (negative && ns > 0) {
       ms += Math.floor(ns / 1e6);
-      ns = 1e6 - (ns % 1e6);
+      ns = ns % 1e6;
```

Comment:
This is wrong!

`negative` indicates that the Absolute is before epoch. With this approach `ns` whill always be >= 0 when it should be -= 0 when `ms` < 0;

Reply:
The problem here is ordering. We need to add both `ms` and `ns`, then determine `negative` and then do the correction as before `1e6 - (ns % 1e6)`;